pkg/handler/eventhandler.go

```diff
@@ -111,7 +113,22 @@ type TypedFuncs[object any, request comparable] struct {
 // Create implements EventHandler.
 func (h TypedFuncs[object, request]) Create(ctx context.Context, e event.TypedCreateEvent[object], q workqueue.TypedRateLimitingInterface[request]) {
 	if h.CreateFunc != nil {
-		h.CreateFunc(ctx, e, q)
+		if reflect.TypeFor[request]() != reflect.TypeOf(reconcile.Request{}) || !reflect.TypeFor[object]().Implements(reflect.TypeFor[client.Object]()) {
+			h.CreateFunc(ctx, e, q)
+		}
+
+		wq, ok := q.(workqueue.TypedRateLimitingInterface[reconcile.Request])
+		if ok {
+			evt, ok := any(e).(event.TypedCreateEvent[client.Object])
+			if ok {
+				item := reconcile.Request{NamespacedName: types.NamespacedName{
```

Comment:
This isn't quite correct, because we need to run `CreateFunc` to know what `item` looks like, here we basically hardcode the same behavior as `EnqueueRequestForObjec` - Lets maybe leave the `TypedFuncs` out as well of this PR?